### DISCUSSION TITLE
Icons

### DISCUSSION THREAD

**Discussion by zaeemali272** [Naezr/ShyFox#126] [Ideas]
Could you add icons for:
Midnight Lizard - https://addons.mozilla.org/en-US/firefox/addon/midnight-lizard-quantum
Auto Tab Discard - https://addons.mozilla.org/en-US/firefox/addon/auto-tab-discard
Video DownloadHelper - https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper

and can you tell me how you make them so i could make them myself.
: )

**Comment by Naezr** [comment]
I'll give it a try. The first two should be easy. I'm not sure about Video Download Helper, though. I don't think I can find a similar icon

> and can you tell me how you make them so i could make them myself.

I found all the icons online. Just looking for something like "bitwarden svg icon". If I can't find one, I look for a replacement. For example moon icon for Dark Reader. The Privacy Badger icon was colored and had a bunch of layers, so I removed all but one layer using Inkscape. And most importantly: how to make the icon adaptive so that it adjusts to the theme colors. You have to open the icon as a text file and replace the value of the `fill` parameter with `context-fill`. But this doesn't always work. Some icons are arranged differently, but in any case there will be a parameter responsible for the color. It should be replaced by `context-fill`.

**Reply by zaeemali272** [reply]
do i have to make it white?
and how do i open the icon as a text file?
![bitwarden](https://github.com/user-attachments/assets/672b6e39-5030-45d7-a6b6-8db0f297a2f8)

**Reply by Naezr** [reply]
> do i have to make it white?

Doesn't matter

> and how do i open the icon as a text file?

Right click -> open with

**Comment by Naezr** [comment]
We already have a power icon. I made a Midnight Lizard icon out of a lizard icon and braces. And the Video download helper is really weird. Its icon simply cannot be monochrome. Any ideas what to replace it with? I don't have any ideas

![image](https://github.com/user-attachments/assets/a03bdecd-3c10-4100-b08f-ef433afa8c91)
![image](https://github.com/user-attachments/assets/73be2736-601b-4e9e-807f-53622bd4c2fb)

**Reply by zaeemali272** [reply]
![image](https://github.com/user-attachments/assets/b8cb061a-ff45-4510-9a09-8a92ebb12e3c)
you can just add three circles

**Comment by zaeemali272** [comment]
bro i just freestyled one in inkscape.
![bitwarden](https://github.com/user-attachments/assets/db2b82ae-c9d2-49bb-b42d-b84237c1f567)
now how do i replace the value of the fill parameter with context-fill 
here:
<svg
   version="1.1"
   id="svg1"
   width="103"
   height="122"
   viewBox="0 0 103 122"
   sodipodi:docname="bitwarden.png"
   inkscape:version="1.3.2 (091e20ef0f, 2023-11-25)"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:svg="http://www.w3.org/2000/svg">
  <defs
     id="defs1" />
  <sodipodi:namedview
     id="namedview1"
     pagecolor="#ffffff"
     bordercolor="#000000"
     borderopacity="0.25"
     inkscape:showpageshadow="2"
     inkscape:pageopacity="0.0"
     inkscape:pagecheckerboard="0"
     inkscape:deskcolor="#d1d1d1"
     showgrid="false"
     inkscape:zoom="4.9917265"
     inkscape:cx="52.687181"
     inkscape:cy="56.092817"
     inkscape:window-width="1366"
     inkscape:window-height="768"
     inkscape:window-x="0"
     inkscape:window-y="0"
     inkscape:window-maximized="1"
     inkscape:current-layer="g1" />
  <g
     inkscape:groupmode="layer"
     inkscape:label="Image"
     id="g1">
    <path
       id="path2"
       style="display:inline;fill:#808080"
       d="m 50.81632,24.314883 -0.02777,81.483527 c 0,0 24.360245,-11.416607 32.211134,-30.10339 L 82.900434,24.420916 Z M 50.397965,118.97741 C 12.015507,102.50913 6.9322945,76.037716 6.9322945,76.037716 c 0,0 0.3075429,-46.892024 -0.078352,-55.804625 -0.3858949,-8.9126 6.8888515,-7.161873 6.8888515,-7.161873 l 75.177535,0.207046 c 0,0 5.648771,-1.068773 4.903884,4.207336 l 0.07515,56.313148 c 0,0 -7.227946,31.331392 -43.501403,45.178662 z" />
  </g>
</svg>

**Reply by Naezr** [reply]
First of all, it's better to save as a regular svg rather than Inkscape svg. And secondly, I'm not exactly sure, but it's either pagecolor or border-color.

**Comment by Naezr** [comment]
![image](https://github.com/user-attachments/assets/b3d28ddc-bf99-4ed9-8876-3c8b991a0f32)

**Reply by zaeemali272** [reply]
i think i did it now where do i put the png

**Reply by Naezr** [reply]
Not png, but svg. Png is not suitable because it cannot change the fill color. Check out this commit: https://github.com/Naezr/ShyFox/commit/b9bfb3908e9199b56ec4e7074f49478609d16531 You need to put the svg icon in the `icons` folder. And add a css style similar to the others, replacing the id of the extension and the path to the icon
